jetty-webapp/src/main/java/org/eclipse/jetty/webapp/WebInfConfiguration.java

```diff
@@ -810,6 +813,29 @@ public static String getCanonicalNameForWebAppTmpDir(WebAppContext context)
 
         return StringUtil.sanitizeFileSystemName(canonicalName.toString());
     }
+    
+    private static String getResourceBasePath(Resource resource)
+    {
+        String tmp = "";
+        try
+        {
+            tmp = URIUtil.decodePath(resource.getURI().getPath());
```

Comment:
Because the two lines are very similar, at the very least we need a comment to say why we are trying them both.
However, I don't like generating a NPE and catching it... I'd prefer to do a null check for the return value and if null try the other.